---
title: "Re: occasional timeout when deleting key on multi-node Riak 1.4"
description: ""
project: community
lastmod: 2014-10-02T10:01:29-07:00
sitemap:
  priority: 0.2
layout: mailinglistitem
mailinglist_id: "msg14976"
mailinglist_parent_id: "msg14974"
author_name: "Igor Senderovich"
project_section: "mailinglistitem"
sent_date: 2014-10-02T10:01:29-07:00
---


There are no other errors in any of the logs at exactly the same time but
there are periodic errors in error.log and console.log of the following
form (and these occurred seconds before and after the crash):

2014-10-02 12:07:57.623 [error] &lt;0.30125.18&gt; CRASH REPORT Process
&lt;0.30125.18&gt; with 10 neighbours exited with reason: no case clause matching
"immediate" in riak\_kv\_vnode:do\_delete/3 line 1321 in gen\_f
sm:terminate/7 line 611
2014-10-02 12:07:57.623 [error] &lt;0.30127.18&gt; Supervisor
{&lt;0.30127.18&gt;,poolboy\_sup} had child riak\_core\_vnode\_worker started with
riak\_core\_vnode\_worker:start\_link([{worker\_module,riak\_core\_vnode\_worker
},{worker\_args,[228359630832953580969325755111919221821239459840,...]},...])
at undefined exit with reason no case clause matching "immediate" in
riak\_kv\_vnode:do\_delete/3 line 1321 in context shutdown
\_error
2014-10-02 12:07:57.623 [error] &lt;0.30127.18&gt; gen\_server &lt;0.30127.18&gt;
terminated with reason: no case clause matching "immediate" in
riak\_kv\_vnode:do\_delete/3 line 1321



crash.log contains:

2014-10-02 12:07:57 =ERROR REPORT====
\*\* State machine &lt;0.30125.18&gt; terminating
\*\* Last message in was
{'EXIT',&lt;0.30124.18&gt;,{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_v
node.erl"},{line,299}]},{gen\_fsm,handle\_msg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}}
\*\* When State == ready
\*\* Data ==
{state,&lt;0.30127.18&gt;,{[&lt;0.30129.18&gt;,&lt;0.30128.18&gt;,&lt;0.30138.18&gt;,&lt;0.30137.18&gt;],[&lt;0.30130.18&gt;,&lt;0.30131.18&gt;,&lt;0.30132.18&gt;,&lt;0.30134.18&gt;,&lt;0.30135.18&gt;,&lt;0.30136.18&gt;]},{[],[]},295960692,10,0,0}
\*\* Reason for termination =
\*\*
{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_vnode.erl"},{line,299}]},{gen\_fsm,handle\_m
sg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}
2014-10-02 12:07:57 =CRASH REPORT====
 crasher:
 initial call: poolboy:init/1
 pid: &lt;0.30125.18&gt;
 registered\_name: []
 exception exit:
{{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_vnode.erl"},{line,299}]}
,{gen\_fsm,handle\_msg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]},[{gen\_fsm,terminate,7,[{file,"gen\_fsm.erl"},{line,611}]},{proc\_lib,init\_p\_do\_
apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}
 ancestors:
[&lt;0.30124.18&gt;,&lt;0.30117.18&gt;,riak\_core\_vnode\_sup,riak\_core\_sup,&lt;0.157.0&gt;]
 messages: []
 links:
[&lt;0.30134.18&gt;,&lt;0.30136.18&gt;,&lt;0.30137.18&gt;,&lt;0.30138.18&gt;,&lt;0.30135.18&gt;,&lt;0.30129.18&gt;,&lt;0.30131.18&gt;,&lt;0.30132.18&gt;,&lt;0.30130.18&gt;,&lt;0.30127.18&gt;,&lt;0.30128.18&gt;]
 dictionary: []
 trap\_exit: true
 status: running
 heap\_size: 987
 stack\_size: 24
 reductions: 403
 neighbours:
 neighbour:
[{pid,&lt;0.30128.18&gt;},{registered\_name,[]},{initial\_call,{riak\_core\_vnode\_worker,init,['Argument\_\_1']}},{current\_function,{gen\_server,loop,6}},{ancestors,[&lt;0.30127.18&gt;,&lt;0.30125.18&gt;,&lt;0.3012
4.18&gt;,&lt;0.30117.18&gt;,riak\_core\_vnode\_sup,riak\_core\_sup,&lt;0.157.0&gt;]},{messages,[]},{links,[&lt;0.30127.18&gt;,&lt;0.30125.18&gt;]},{dictionary,[]},{trap\_exit,false},{status,waiting},{heap\_size,610},{stack\_size,9},{red
uctions,526}]
 neighbour:
[{pid,&lt;0.30130.18&gt;},{registered\_name,[]},{initial\_call,{riak\_core\_vnode\_worker,init,['Argument\_\_1']}},{current\_function,{gen\_server,loop,6}},{ancestors,[&lt;0.30127.18&gt;,&lt;0.30125.18&gt;,&lt;0.3012
4.18&gt;,&lt;0.30117.18&gt;,riak\_core\_vnode\_sup,riak\_core\_sup,&lt;0.157.0&gt;]},{messages,[]},{links,[&lt;0.30127.18&gt;,&lt;0.30125.18&gt;]},{dictionary,[]},{trap\_exit,false},{status,waiting},{heap\_size,233},{stack\_size,9},{red
uctions,27}]
...
2014-10-02 12:07:57 =SUPERVISOR REPORT====
 Supervisor: {&lt;0.30127.18&gt;,poolboy\_sup}
 Context: shutdown\_error
 Reason:
{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_vnode.erl"},{line,299}]},{ge
n\_fsm,handle\_msg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}
 Offender:
[{nb\_children,10},{name,riak\_core\_vnode\_worker},{mfargs,{riak\_core\_vnode\_worker,start\_link,[[{worker\_module,riak\_core\_vnode\_worker},{worker\_args,[22835963083295358096932575511191922182
1239459840,[],worker\_props,&lt;0.30124.18&gt;]},{worker\_callback\_mod,riak\_kv\_worker},{size,10},{max\_overflow,0}]]}},{restart\_type,temporary},{shutdown,5000},{child\_type,worker}]

2014-10-02 12:07:57 =ERROR REPORT====
\*\* Generic server &lt;0.30127.18&gt; terminating
\*\* Last message in was
{'EXIT',&lt;0.30125.18&gt;,{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_v
node.erl"},{line,299}]},{gen\_fsm,handle\_msg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}}
\*\* When Server state ==
{state,{&lt;0.30127.18&gt;,poolboy\_sup},simple\_one\_for\_one,[{child,undefined,riak\_core\_vnode\_worker,{riak\_core\_vnode\_worker,start\_link,[[{worker\_module,riak\_core\_vnode\_worker},{worker
\_args,[228359630832953580969325755111919221821239459840,[],worker\_props,&lt;0.30124.18&gt;]},{worker\_callback\_mod,riak\_kv\_worker},{size,10},{max\_overflow,0}]]},temporary,5000,worker,[riak\_core\_vnode\_worker]}
],{set,10,16,16,8,80,48,{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]},{{[&lt;0.30131.18&gt;],[&lt;0.30132.18&gt;],[],[&lt;0.30134.18&gt;],[&lt;0.30135.18&gt;],[&lt;0.30136.18&gt;],[&lt;0.30137.18&gt;],[&lt;0.30138.18&gt;],[],[],[],[],[],[&lt;
0.30128.18&gt;],[&lt;0.30129.18&gt;],[&lt;0.30130.18&gt;]}}},0,1,[],poolboy\_sup,{riak\_core\_vnode\_worker,[{worker\_module,riak\_core\_vnode\_worker},{worker\_args,[228359630832953580969325755111919221821239459840,[],worker
\_props,&lt;0.30124.18&gt;]},{worker\_callback\_mod,riak\_kv\_worker},{size,10},{max\_overflow,0}]}}
\*\* Reason for termination ==
\*\*
{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_vnode.erl"},{line,299}]},{gen\_fsm,handle\_m
sg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}
2014-10-02 12:07:57 =CRASH REPORT====
 crasher:
 initial call: supervisor:poolboy\_sup/1
 pid: &lt;0.30127.18&gt;
 registered\_name: []
 exception exit:
{{{case\_clause,"immediate"},[{riak\_kv\_vnode,do\_delete,3,[{file,"src/riak\_kv\_vnode.erl"},{line,1321}]},{riak\_core\_vnode,vnode\_command,3,[{file,"src/riak\_core\_vnode.erl"},{line,299}]}
,{gen\_fsm,handle\_msg,7,[{file,"gen\_fsm.erl"},{line,494}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]},[{gen\_server,terminate,6,[{file,"gen\_server.erl"},{line,747}]},{proc\_lib,init
\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,227}]}]}
 ancestors:
[&lt;0.30125.18&gt;,&lt;0.30124.18&gt;,&lt;0.30117.18&gt;,riak\_core\_vnode\_sup,riak\_core\_sup,&lt;0.157.0&gt;]
 messages: []
 links: []
 dictionary: []
 trap\_exit: true
 status: running
 heap\_size: 4181
 stack\_size: 24
 reductions: 1530
 neighbours:


On Thu, Oct 2, 2014 at 12:20 PM, Dmitri Zagidulin 
wrote:

&gt; Thanks. Are there entries in any of the other logs? (like the crash dump).
&gt;
&gt; Can you also provide more info on the nodes themselves. What size AWS
&gt; instances are you running? Is the delete timeout happening while load
&gt; testing?
&gt;
&gt; On Thu, Oct 2, 2014 at 12:11 PM, Igor Senderovich &lt;
&gt; isenderov...@esperdyne.com&gt; wrote:
&gt;
&gt;&gt; Thanks for your help, Dmitri,
&gt;&gt;
&gt;&gt; I get the following in error.log:
&gt;&gt; 2014-10-02 12:05:45.037 [error] &lt;0.6359.19&gt; Webmachine error at path
&gt;&gt; "/buckets/imc/keys/5134a18660494ea5553d2c90ef9eea2f" : "Service Unavailable"
&gt;&gt;
&gt;&gt; And no, there is no load balancer on our cluster.
&gt;&gt; Thank you
&gt;&gt;
&gt;&gt;
&gt;&gt; On Thu, Oct 2, 2014 at 11:52 AM, Dmitri Zagidulin 
&gt;&gt; wrote:
&gt;&gt;
&gt;&gt;&gt; One other question - are you using a load balancer for your cluster
&gt;&gt;&gt; (like HAProxy or the like). In which case, take a look at its logs, also.
&gt;&gt;&gt;
&gt;&gt;&gt; On Thu, Oct 2, 2014 at 11:51 AM, Dmitri Zagidulin 
&gt;&gt;&gt; wrote:
&gt;&gt;&gt;
&gt;&gt;&gt;&gt; Igor,
&gt;&gt;&gt;&gt; Can you look in the riak log directory, in the error.log (and console
&gt;&gt;&gt;&gt; log and crash dump file) to see if there's any entries, around the time of
&gt;&gt;&gt;&gt; the delete operation? And post them here?
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt; On Thu, Oct 2, 2014 at 11:45 AM, Igor Senderovich &lt;
&gt;&gt;&gt;&gt; isenderov...@esperdyne.com&gt; wrote:
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt; Hi,
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt; I get a timeout when deleting a key, reproducible in about 1 in 10
&gt;&gt;&gt;&gt;&gt; times:
&gt;&gt;&gt;&gt;&gt; $ curl -i -vvv
&gt;&gt;&gt;&gt;&gt; http://myhost:8098/buckets/imc/keys/5134a18660494ea5553d2c90ef9eea2f
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt; \* About to connect() to dp1.prod6.ec2.cmg.net port 8098
&gt;&gt;&gt;&gt;&gt; \* Trying 10.12.239.90... connected
&gt;&gt;&gt;&gt;&gt; \* Connected to dp1.prod6.ec2.cmg.net (10.12.239.90) port 8098
&gt;&gt;&gt;&gt;&gt; &gt; DELETE /buckets/imc/keys/5134a18660494ea5553d2c90ef9eea2f HTTP/1.1
&gt;&gt;&gt;&gt;&gt; &gt; User-Agent: curl/7.15.5 (x86\_64-redhat-linux-gnu) libcurl/7.15.5
&gt;&gt;&gt;&gt;&gt; OpenSSL/0.9.8b zlib/1.2.3 libidn/0.6.5
&gt;&gt;&gt;&gt;&gt; &gt; Host: dp1.prod6.ec2.cmg.net:8098
&gt;&gt;&gt;&gt;&gt; &gt; Accept: \*/\*
&gt;&gt;&gt;&gt;&gt; &gt;
&gt;&gt;&gt;&gt;&gt; &lt; HTTP/1.1 503 Service Unavailable
&gt;&gt;&gt;&gt;&gt; HTTP/1.1 503 Service Unavailable
&gt;&gt;&gt;&gt;&gt; &lt; Server: MochiWeb/1.1 WebMachine/1.10.0 (never breaks eye contact)
&gt;&gt;&gt;&gt;&gt; Server: MochiWeb/1.1 WebMachine/1.10.0 (never breaks eye contact)
&gt;&gt;&gt;&gt;&gt; &lt; Date: Wed, 01 Oct 2014 16:11:41 GMT
&gt;&gt;&gt;&gt;&gt; Date: Wed, 01 Oct 2014 16:11:41 GMT
&gt;&gt;&gt;&gt;&gt; &lt; Content-Type: text/plain
&gt;&gt;&gt;&gt;&gt; Content-Type: text/plain
&gt;&gt;&gt;&gt;&gt; &lt; Content-Length: 18
&gt;&gt;&gt;&gt;&gt; Content-Length: 18
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt; request timed out
&gt;&gt;&gt;&gt;&gt; \* Connection #0 to host dp1.prod6.ec2.cmg.net left intact
&gt;&gt;&gt;&gt;&gt; \* Closing connection #0
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt; This is on Riak 1.4 on a 5 node cluster with an n-value of 3.
&gt;&gt;&gt;&gt;&gt; Thank you for your help
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt; \_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
&gt;&gt;&gt;&gt;&gt; riak-users mailing list
&gt;&gt;&gt;&gt;&gt; riak-users@lists.basho.com
&gt;&gt;&gt;&gt;&gt; http://lists.basho.com/mailman/listinfo/riak-users\_lists.basho.com
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;
&gt;&gt;
&gt;
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
riak-users mailing list
riak-users@lists.basho.com
http://lists.basho.com/mailman/listinfo/riak-users\_lists.basho.com

